tests/test_daemon_clone.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "daemon_test_fixture.h"
+
+#include "common.h"
+#include "mock_platform.h"
+#include "mock_server_reader_writer.h"
+#include "mock_settings.h"
+#include "mock_virtual_machine.h"
+#include "mock_vm_image_vault.h"
+
+#include <src/daemon/daemon.h>
+
+#include <multipass/constants.h>
+#include <multipass/format.h>
+
+namespace mp = multipass;
+namespace mpt = multipass::test;
+using namespace testing;
+
+struct TestDaemonClone : public mpt::DaemonTestFixture
+{
+    void SetUp() override
+    {
+        config_builder.vault = std::make_unique<NiceMock<mpt::MockVMImageVault>>();
+    }
+
+    auto build_daemon_with_mock_instance()
+    {
+        auto instance_unique_ptr = std::make_unique<NiceMock<mpt::MockVirtualMachine>>(mock_src_instance_name);
+        auto* instance_raw_ptr = instance_unique_ptr.get();
+
+        EXPECT_CALL(mock_factory, create_virtual_machine).WillOnce(Return(std::move(instance_unique_ptr)));
+
+        const auto [temp_dir, _] = plant_instance_json(fake_json_contents(mac_addr, extra_interfaces));
+        config_builder.data_directory = temp_dir->path();
+        auto daemon = std::make_unique<mp::Daemon>(config_builder.build());
+
+        return std::pair{std::move(daemon), instance_raw_ptr};
+    }
+
+    const std::string mock_src_instance_name{"real-zebraphant"};
+    const std::string mac_addr{"52:54:00:73:76:28"};
+    std::vector<mp::NetworkInterface> extra_interfaces;
+
+    const mpt::MockVirtualMachineFactory& mock_factory = *use_a_mock_vm_factory();
+
+    const mpt::MockPlatform::GuardedMock attr{mpt::MockPlatform::inject<NiceMock>()};
+    const mpt::MockPlatform& mock_platform = *attr.first;
+
+    const mpt::MockSettings::GuardedMock mock_settings_injection = mpt::MockSettings::inject<NiceMock>();
+};
+
+TEST_F(TestDaemonClone, missingOnSrcInstance)
+{
+    constexpr std::string_view src_instance_name = "non_exist_instance";
```

Comment:
There is no such thing as  `constexpr std::string`, it only can be runtime. The compile time string types normally are `std::string_view` or `char* `. 

The real comparison here you mean maybe is
```
constexpr std::string_view src_instance_name = "non_exist_instance";
request.set_source_name(std::string{src_instance_name});
```
vs 

`request.set_source_name("non_exist_instance");`

Both store the string literal in the static storage of the program and get it copied into std::string. The first one has `std::string_view` points to it. The second one has `char*` points to that and `std::string`'s constructor implicitly constructs it into `std::string`. So it is the same in terms of making a copy. 

I thought there was a small gain because `std::string_view` knows the size so the copy is faster. But later I dug deeper and it turns out that the compliers know the the size of the char* string literal. So these two are more or less equivalent. I will change to the 2nd one.

Reply:
If you mean `const std::string src_instance_name = "non_exist_instance";` That would be the same in terms of copy. But it passes a pointer(reference) to the `request.set_source_name()`

Reply:
I'm just seeing that you're creating a `std::string` anyways. Because of that, I would personally just use `std::string` instead of `string_view` and not have to worry about types, creating strings later which just create clutter in the code.

Reply:
This is a minor thing in terms of choosing which one. I do not mind changing to `const std::string src_instance_name = "non_exist_instance"` to save the explicit construction syntax.